pymc3/distributions/distribution.py

```diff
@@ -115,13 +127,180 @@ def logcdf(op, var, rvs_to_values, *dist_params, **kwargs):
         return new_cls
 
 
+def _valid_ellipsis_position(items: Union[None, Shape, Dims, Size]) -> bool:
```

Comment:
I wonder if with all these changes we should places these into a `shape.py`.

Reply:
Then, the extra logic below in `__new__ ` and `dist()` could go in there too and everything would be nice and tidy.

Reply:
Let's not get distracted by "hiding everything under the carpet". There is actually very little extra logic left in these methods and extracting it out does not simplify anything.
Adding more and more layers of function calls doesn't solve any of our actual problems: A lot of these data types around here are way to promiscuous.

Unless someone points out exactly which block of code should be extracted and why, I have more useful things on my to do list.

Reply:
These functions are now split and tested individually.